src/DatePicker/CalendarMonth.js

```diff
@@ -34,25 +31,27 @@ const CalendarMonth = React.createClass({
 
   _getDayElements(week, i) {
     return week.map((day, j) => {
-      const isSameDate = DateTime.isEqualDate(this.props.selectedDate, day);
-      const disabled = this._shouldDisableDate(day);
-      const selected = !disabled && isSameDate;
+      if (this.props.getDayElement) {
+        const component = this.props.getDayElement(day, this.props.selectedDate);
 
-      if (isSameDate) {
-        if (disabled) {
-          this._selectedDateDisabled = true;
-        } else {
-          this._selectedDateDisabled = false;
+        if (!ReactTestUtils.isElementOfType(component, DayButton)) {
+          throw 'getDayElement must return an element of type DayButton';
         }
+
+        return React.cloneElement(component, {
```

Comment:
Lets say you have a custom AvailabilityDatePicker component which renders a DatePicker component and which has a method getAvailabilityDayElement() which is passed as a prop to the DatePicker component and returns each DayButton that is gonna be rendered inside the DatePicker.

The key prop of the DayButton element can be set inside the getAvailabilityDayElement but the onTouchTap event handler is an internal handler of the DatePicker ( and sub-component ) so it can't be set from the AvailabilityDatePicker component.